packages/shared/src/components/auth/RegistrationForm.tsx

```diff
@@ -58,14 +62,39 @@ export const RegistrationForm = ({
 }: RegistrationFormProps): ReactElement => {
   const { trackEvent } = useContext(AnalyticsContext);
   const [isSubmitted, setIsSubmitted] = useState<boolean>(false);
+  const [username, setUsername] = useState<string>('');
+  const [name, setName] = useState<string>('');
   const isAuthorOnboarding = trigger === AuthTriggers.Author;
+  const { requestMethod } = useRequestProtocol();
+  const usernameQueryKey = ['generateUsername', name];
+  const { data: generatedUsername } = useQuery<{
+    generateUniqueUsername: string;
+  }>(
+    usernameQueryKey,
+    () =>
+      requestMethod(
+        graphqlUrl,
+        GET_USERNAME_SUGGESTION,
+        { name },
+        { requestKey: JSON.stringify(usernameQueryKey) },
+      ),
+    { enabled: !!name.length },
+  );
 
   useEffect(() => {
     trackEvent({
       event_name: AuthEventNames.StartSignUpForm,
     });
   }, []);
 
+  useEffect(() => {
+    if (username.length) return;
+
+    if (generatedUsername?.generateUniqueUsername) {
+      setUsername(generatedUsername.generateUniqueUsername);
+    }
+  }, [generatedUsername]);
```

Comment:
This isn't for the social login, this is only used to set the state/value of the username field once we get a result returned, hence I am unsure how this could be done on the API side?

Reply:
Oh right, could we not simply call `setUsername` from the return value?
Seems like we can avoid the useEffect here.

Reply:
Yes this wasn't something i was aware we could do, this has now be fixed in the next commit